src/codecs/tiff.rs

```diff
@@ -310,6 +339,14 @@ fn u8_slice_as_u16(buf: &[u8]) -> ImageResult<&[u16]> {
     })
 }
 
+fn u8_slice_as_f32(buf: &[u8]) -> ImageResult<&[f32]> {
+    bytemuck::try_cast_slice(buf).map_err(|err| {
+        ImageError::Parameter(ParameterError::from_kind(ParameterErrorKind::Generic(
```

Comment:
4-byte alignment is required only for RGB32 images (which are being added in), does that break the encoder interface?

Reply:
Yes. Other parts of the crate allow 1-byte aligned buffers for storing 16-bit integers or 32-bit floats. For instance, the HDR decoder which decodes f32's into a byte slice.

The reason we allow it is that you can't easily create an over-aligned `Vec<u8>` and we don't want to _force_ users to store image data in typed slices before passing them to encoders/decoders. And it would be too easy of a gotcha to overlook when using the API